### PR TITLE
executor, plan: support analyze partition table

### PR DESCRIPTION
**Pull request by lamxTyler**
## What have you changed? (mandatory)

Implement `Analyze` for the partition table, analyze and store the stats for each partition.

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit test.
## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.
## Does this PR affect tidb-ansible update? (mandatory)

No.
## Does this PR need to be added to the release notes? (mandatory)

No.

## Add a few positive/negative examples (optional)

PTAL @coocood @zz-jason @winoros @tiancaiamao

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: statistics/builder.go [hunk 1]
```diff
@@ -156,10 +156,10 @@ func BuildColumn(ctx sessionctx.Context, numBuckets, id int64, collector *Sample
 
 // AnalyzeResult is used to represent analyze result.
 type AnalyzeResult struct {
-	TableID int64
-	Hist    []*Histogram
-	Cms     []*CMSketch
-	Count   int64
-	IsIndex int
-	Err     error
+	PhysicalID int64
```
Comment (shenli) [inline statistics/builder.go]: What is the PhysicalID?

Comment (lamxTyler) [reply statistics/builder.go]: It is the partition id for a partitioned table, otherwise, it is the table id.

Comment (shenli) [reply statistics/builder.go]: Please add a comment for it.

File: executor/analyze.go [hunk 2]
```diff
@@ -229,7 +229,7 @@ func analyzeColumnsPushdown(colExec *AnalyzeColumnsExec) statistics.AnalyzeResul
 // AnalyzeColumnsExec represents Analyze columns push down executor.
 type AnalyzeColumnsExec struct {
 	ctx           sessionctx.Context
-	tblInfo       *model.TableInfo
+	physicalID    int64
```
Comment (zz-jason) [inline executor/analyze.go]: we'd better add a comment for this field.

File: plan/common_plans.go [hunk 3]
```diff
@@ -355,15 +355,15 @@ type Delete struct {
 
 // AnalyzeColumnsTask is used for analyze columns.
 type AnalyzeColumnsTask struct {
-	TableInfo *model.TableInfo
-	PKInfo    *model.ColumnInfo
-	ColsInfo  []*model.ColumnInfo
+	PhysicalID int64
```
Comment (zz-jason) [inline plan/common_plans.go]: ditto

**Comment by lamxTyler** [comment]
PTAL @shenli @zz-jason

**Pull request merged**